tiledb/cloud/groups.py

```diff
@@ -0,0 +1,181 @@
+    :param credentials_name: The name of the storage credential to use for
+        creating the group. If not provided, uses the namespace's default
+        credential for groups.
+    """
+    if not (namespace and storage_uri and credentials_name):
+        namespace, default_path, default_cred = _default_ns_path_cred(namespace)
+        storage_uri = storage_uri or (default_path + "/" + name)
+        credentials_name = credentials_name or default_cred
+
+    groups_client = client.build(api_v2.GroupsApi)
+    groups_client.create_group(
+        group_namespace=namespace,
+        x_tiledb_cloud_access_credentials_name=credentials_name,
+        group_creation=api_v2.GroupCreationRequest(
+            group_details=api_v2.GroupCreationRequestGroupDetails(
+                name=name,
+                uri=storage_uri,
+            )
+        ),
+    )
+    if parent_uri:
+        _add_to(namespace=namespace, name=name, parent_uri=parent_uri)
+
+
+def register(
+    storage_uri: str,
+    *,
+    name: Optional[str] = None,
+    namespace: Optional[str] = None,
+    credentials_name: Optional[str] = None,
+    parent_uri: Optional[str] = None,
+):
+    """Registers a pre-existing group."""
+    if not (namespace and credentials_name):
+        namespace, _, default_cred = _default_ns_path_cred(namespace)
+        credentials_name = credentials_name or default_cred
+
+    if not name:
+        # Extract the basename from the storage URI and use it for the name.
+        parsed_uri = urllib.parse.urlparse(storage_uri)
+        name = posixpath.basename(parsed_uri.path)
+    groups_client = client.build(api_v2.GroupsApi)
+    groups_client.register_group(
+        group_namespace=namespace,
+        x_tiledb_cloud_access_credentials_name=credentials_name,
+        group_registration=api_v2.GroupRegistrationRequest(
+            group_details=api_v2.GroupRegistrationRequestGroupDetails(
+                name=name,
+                uri=storage_uri,
+                access_credentials_name=credentials_name,
+                parent=parent_uri,
+            )
+        ),
+    )
+    if parent_uri:
+        _add_to(namespace=namespace, name=name, parent_uri=parent_uri)
+
+
+def deregister(
+    uri: str,
+    *,
+    recursive: bool = False,
+) -> None:
+    """Deregisters the given group from TileDB Cloud.
+
+    :param uri: The URI of the group to deregister.
+    :param recursive: If true, deregister the group recursively by deregistering
+        all of the elements of the group (and all elements of those groups,
+        recursively) before deregistering the group itself.
+    """
+    namespace, name = utils.split_uri(uri)
+
+    array_api = client.build(rest_api.ArrayApi)
+    groups_api = client.build(api_v2.GroupsApi)
+    groups_api_v1 = client.build(rest_api.GroupsApi)
+    if recursive:
+        while True:
+            contents: rest_api.GroupContents = groups_api_v1.get_group_contents(
```

Comment:
It might be easier to use the tiledb group function:
```
with tiledb.Group(uri, ctx=tiledb.cloud.Ctx() as G:
   for member in G:
      memory.uri
```

Reply:
At this point I would prefer to use *only* direct use of the HTTP API rather than mixing the HTTP API with using the local TileDB Core APIs, but it might make sense to do this in the future.